File: multiples_3_or_5/go/solution.go
package kata

func Multiple3And5(number int) int {
	var sum int
	for i := 0; i < number; i++ {
		if isMultiple3or5(i) {
			sum += i
		}
	}
	return sum
}

func isMultiple3or5(number int) bool {
	var result bool
	var mod_3 = number % 3
	var mod_5 = number % 5

	if mod_3 == 0 || mod_5 == 0 {
		result = true
	} else {
		result = false
	}
	return result
}

File: multiples_3_or_5/python/solution.py
""" Multiples of 3 or 5 

Solution I
Use a list comprehension to filter all that are not dividable by 3 or 5 
sum up the list 
"""
import logging


def solution(number: int) -> int:
    """Find the sum of numbers which are lower than the input
    and multiples of 3 or 5
    """
    logging.debug(f"Input was {number}")
    multiples = [x for x in range(number) if is_multiple_of_3_or_5(x)]
    logging.debug(f"Identified {multiples} as multiples")
    result = sum(multiples)
    logging.info(f"Sum is {result}")
    return result


def is_multiple_of_3_or_5(x: int) -> bool:
    """ 
    Return if the number is divisible by 3 or 5 
    """
    is_multiple = x % 3 == 0 or x % 5 == 0
    logging.debug(f"Is {x} a multiple: {is_multiple}")
    return is_multiple
